src/modules/item/sagas.ts

```diff
@@ -153,16 +156,43 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     }
   }
 
+  function* fetchCollectionItemsWithBatch(collectionId: string, pagesToFetch: number[], limit?: number, isReviewing = false) {
+    const REQUEST_BATCH_SIZE = 10
+    const queue = new PQueue({ concurrency: REQUEST_BATCH_SIZE })
```

Comment:
Do you think there's room for an abstraction here? like `concurrent(size, (batch) => { /* push stuff, return batch */ })`

to reuse in the cases we're using PQueue. Can be done in a separate issue

Reply:
Sure, #1928 as a follow up.